SparkPost/2web2ui
# SA-1408: Fix clearing subaccount on HS page bug

Note: Give your PR a short title. Something like "_TICKET NUMBER: feature description_" is good.

### What Changed
 - Fixed subaccount filter handler to accept null values and revert to default.
 - Added selectedItem so that the selected subaccount shows in the typeahead.

### How To Test

1. Enable "Hibana" theme
2. On any of the Signals overview pages (e.g. /signals/health-score), open subaccount select
3. Pick "Search for Subaccount"
4. Select a subaccount with typeahead
5. Confirm loads correctly
6. Then open "Search for Subaccount" and click "Clear"
7. No errors should appear
